test/EventListener/JsonApiProblemExceptionListenerTest.php

```diff
@@ -88,18 +86,23 @@ public function it_parses_an_api_problem_response(): void
         $this->request->getRequestFormat()->willReturn('json');
         $this->request->getContentType()->willReturn('application/json');
 
-        $this->event->setResponse(Argument::that(function (JsonResponse $response) {
-            return 500 === $response->getStatusCode()
-                && 'application/problem+json' === $response->headers->get('Content-Type')
-                && $response->getContent() === json_encode([
-                    'status' => 500,
-                    'type' => HttpApiProblem::TYPE_HTTP_RFC,
-                    'title' => HttpApiProblem::getTitleForStatusCode(500),
-                    'detail' => 'error',
-                ]);
-        }))->shouldBeCalled();
-
-        $listener->onKernelException($this->event->reveal());
+        $this->event->setResponse(new JsonResponse([
```

Comment:
The listener should be setting this response, not the unit test. So this might lead to false positives.
The original code was validating the content of the response, not setting it.
Validation on the header is missing.

Reply:
Fixed, works fine without the setResponse